hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/preloader/NearCachePreloader.java

```diff
@@ -224,7 +224,7 @@ private int loadKeySet(BufferingInputStream bis, DataStructureAdapter<Object, ?>
                 break;
             }
             Data key = new HeapData(payload);
-            builder.add(serializeKeys ? key : serializationService.toObject(key));
+            builder.add(serializationService.toObject(key));
```

Comment:
I thought about adding code to the `getAll()` method to deal with serialized keys as input as well. But this would make the proxy code even more complex. Since the preloader works in its own thread I went this way for now. Maybe we can have a small (internal) method which don't even return the results, just populates the Near Cache and can handle serialized keys.